File: Data Structures & Algorithms/two-integer-sum/submission-0.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        // Solution: Brute force solution with a double for loop checking
        // ith + jth (i+1) position and checking if it equals the target
        // then returning [i,j]
        // O(n^2) time | O(n) space

        var result = [Int]()

        for i in 0..<nums.count {
            for j in i+1..<nums.count {
                if (nums[i]+nums[j]) == target {
                    result.append(i)
                    result.append(j)
                    
                    return result
                }
            }
        }
        return result

        //-----------------------------------------------------------------

        // Solution: 
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        hm = {}

        for i in range(len(numbers)):
            diff = target - numbers[i]

            if diff not in hm:
                hm[numbers[i]] = i
            else:
                return [hm[diff] + 1, i + 1]
        
        return []